File: vecfleet/Api/models/Vehiculo.cs
using System;
using System.Collections.Generic;

namespace Api.models
{
    public partial class Vehiculo
    {
        public int Id { get; set; }
        public string Marca { get; set; } = null!;
        public int CantidadRuedas { get; set; }
        public string Modelo { get; set; } = null!;
        public string Patente { get; set; } = null!;
        public string Chasis { get; set; } = null!;
        public long KmsRecorrido { get; set; }
        public long KmsProximoMantenimiento { get; set; }
        public int Anio { get; set; }
        public DateTime FechaRegistro { get; set; }
        public int IdTipoVehiculo { get; set; }
    }
}

File: vecfleet.client/src/types/Vehiculo.ts
 interface VehiculoType {
  anio:                    number;
  cantidadRuedas:          number;
  chasis:                  string;
  fechaRegistro:           Date;
  id:                      number;
  idTipoVehiculo:          number;
  kmsProximoMantenimiento: number;
  kmsRecorrido:            number;
  marca:                   string;
  modelo:                  string;
  patente:                 string;
  tipoVehiculo:            TipoVehiculoType;
 }
 
 interface TipoVehiculoType {
  descripcion: string;
  estado:      boolean;
  id:          number;
 }

 interface VehiculoRequestType{
    marca: string;
    cantidadRuedas: number;
    modelo: string;
    patente: string;
    chasis: string;
    kmsRecorrido: number;
    kmsProximoMantenimiento: number;
    anio: number;
    idTipoVehiculo: number;
 }
 export type {VehiculoType, TipoVehiculoType, VehiculoRequestType}